lib/services/database/user_repository_service.dart

```diff
@@ -0,0 +1,43 @@
+import "dart:async";
+
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_data.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/services/database/firestore_service.dart";
+
+/// This repository service is responsible for managing the users in the database
+class UserRepositoryService {
+  final CollectionReference _collectionRef;
+
+  UserRepositoryService({
+    required FirebaseFirestore firestore,
+  }) : _collectionRef = firestore.collection(UserFirestore.collectionName);
+
+  /// This method will retrieve the user with id [uid] from the database
+  Future<UserFirestore> getUser(UserIdFirestore uid) async {
+    final docSnap = await _collectionRef.doc(uid.value).get();
+
+    return UserFirestore.fromDb(docSnap);
+  }
+
+  /// This method will set the user with id [uid] to have the data [userData]
+  /// If the user does not exist yet, it will be created
+  Future<void> setUser(UserIdFirestore uid, UserData userData) async {
```

Comment:
How do we get a `UserIdFirestore` for the logged in user? I am guessing this can be extracted, but I can't seem to find where

Reply:
Thanks for the comment ! This is effectively a point that I have not really well explained.
The firestore collection will be constructed such that the id of the user document is the same as its id from Firebase Auth.
So in order to get the `UserIdFirestore`, we have to get the uid of the current user through Firebase Auth.
Here is a snippet of how we could do it:

```dart
    final firebaszeAuth = FirebaseAuth.instance;
    final userId = firebaszeAuth.currentUser?.uid;
    if (userId == null) {
      // The user is not logged in...
      throw Exception("User is not logged in");
    }
    final userIdFirestore = UserIdFirestore(value: userId);
```

Reply:
I am then curious to know if it would be interesting to add a `loggedInUserIdProvider` in `login_view_model.dart`:
```dart
final loggedInUserIdProvider = StreamProvider<UserIdFirestore?>((ref) {
  return ref.watch(firebaseAuthProvider).authStateChanges().map((user) {
    if (user == null) {
      return null;
    }

    return UserIdFirestore(value: user.uid);
  });
});
```
This code snippet was completely copied of the `userProvider` and thus raises another question in me. Are the classes `Post`, `ProfileData` and `LoginUser` still used? What is the difference between a `LoginUser` and a `UserFirestore`/`UserData`? Should there be any easy way to convert from one to another? Same question for `Post` and `PostFirestore`/`PostData`.

Reply:
Yes it could be a good idea to add `loggedInUserIdProvider` ! This might come useful in the future altough I currently don't have an example in mind of where we could use it. But I think that this might be beyond the scope of this PR as it concerns the view model.

I think that the difference between `LoginUser` and `UserFirestore` is the following.
- `LoginUser` represents the user from the Firebase Auth service with the data that we can get from that (the `uid` and possibly the email) (here we can only get the public data of the google account) 

- `UserFirestore` represents the user as it is stored in our database, so with all the fields that we have added (joinDate, etc.)

Regarding  the classes `Post`, `ProfileData`. I think that those are here because we needed them to allow for the people working on the UI to have a way to represent the data to display while the repositories weren't implemented.
Regarding the fact of if we should keep both or not, I think once again that this should be done in another PR as this is going to imply some modification of the UI code.